scapy/layers/ipsec.py

```diff
@@ -111,7 +113,7 @@ class ESP(Packet):
     name = 'ESP'
 
     fields_desc = [
-        XIntField('spi', 0x0),
+        XIntField('spi', 0xdeadbeef),
```

Comment:
Why this value?

Reply:
It must not be zero by default, because zero is the `non_esp` marker for NAT-traversial. If you have a better suggestion, I'm happy to change it.

Reply:
It was actually a reference to an existing comment:

https://github.com/secdev/scapy/blob/139966f37ee4188c676b958260312b1e4ad81be2/scapy/layers/ipsec.py#L12-L13

Reply:
The only convincing argument against using 'dead beef' is that it is not politically correct. 😉

Reply:
> Why this value?

I'm sorry @gpotter2 , my last answers were probably not very helpful, since I implicitly assumed that everybody has a detailed knowledge of the IKE protocol. The [Security Parameter Index (SPI)](https://www.rfc-editor.org/rfc/rfc4303#section-2.1) is a unique 32-bit value identifying the encrypted channel. It is chosen at random and negotiated by the IKE protocol. In real-world examples, you will alway have to set it explicitly to a negotiated value, otherwise the peer will just ignore your message. 

For dissection, it is only relevant whether the value is zero or not. The arbitrary value `0xdeadbeef` is just serves as a default for toy examples, just like 'Jon Doe'.